docs/src/main/asciidoc/spring-security.adoc

```diff
@@ -0,0 +1,252 @@
+quarkus.security.users.embedded.users.stuart=test
+quarkus.security.users.embedded.roles.stuart=user
+----
+
+The corresponding test could look like:
+
+== GreetingControllerTest
+
+Note that a test for the controller has been created as well:
+
+[source, java]
+----
+package org.acme.spring.web;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+
+@QuarkusTest
+public class GreetingControllerTest {
+
+    @Test
+    public void testHelloEndpointForbidden() {
+        given().auth().preemptive().basic("stuart", "test")
+                .when().get("/greeting")
+                .then()
+                .statusCode(403);
+    }
+
+    @Test
+    public void testHelloEndpoint() {
+        given().auth().preemptive().basic("scott", "jb0ss")
+                .when().get("/greeting")
+                .then()
+                .statusCode(200)
+                .body(is("hello"));
+    }
+
+}
+----
+== Test the changes
+
+Open your browser again to http://localhost:8080/greeting.
+
+The result should be:
+----
+Access to localhost was denied
+You don't have authorization to view this page.
+HTTP ERROR 403
+----
+
+== Run the application as a native executable
+
+You can of course create a native image using the instructions of the link:building-native-image[Building a native executable guide].
+
+
+== Supported Spring Security functionalities
+
+Quarkus currently supports a subset of the functionalities that Spring Security provides. More specifically Quarkus supports the security related features of authorization semantics
+(think of `@Secured` instead of `@RolesAllowed`).
+
+=== Annotations
+
+The table below summarizes the supported annotations:
+
+.Supported Spring Security annotations
+|===
+|Name|Comments
+
+|@Secured
+|
+
+|===
+
+
+== Important Technical Note
+
+Please note that the Spring support in Quarkus does not start a Spring Application Context nor are any Spring infrastructure classes run.
+Spring classes and annotations are only used for reading metadata and / or are used as user code method return types or parameter types.
+What that means for end users, is that adding arbitrary Spring libraries will not have any effect. Moreover Spring infrastructure
+classes (like `org.springframework.beans.factory.config.BeanPostProcessor` for example) will not be executed.
+
+== Conversion Table
+
+The following table shows how Spring Security annotations can be converted to JAX-RS annotations.
+
+|===
+|Spring |JAX-RS |Comments
+
+|@Secured("admin")
+|@RolesAllowed("admin")
+|
+
+|===
+
+== More Spring guides
+
+Quarkus support has more Spring compatibility features. See the following guides for more details:
+
+* link:spring-di[Quarkus - Extension for Spring DI]
+* link:spring-web[Quarkus - Extension for Spring Web]
+* link:spring-data-jpa[Quarkus - Extension for Spring Data JPA]
```

Comment:
We will need to add the Security one in this section of the other Spring guides.

Reply:
Good idea! Best add it to all guides not just the web one as has currently been done